docs/control-plane-architecture.adoc

```diff
@@ -0,0 +1,252 @@
+* Background tasks are continuous processes.  They can provide whatever detailed status they want to, including things like: activity counters, error counters, ringbuffers of recent events, data produced by the task, etc.  These can be viewed with `omdb`.
+
+== Cold start
+
+"Cold start" refers to starting the control plane from a rack that's completely powered off.  Achieving this requires careful consideration of where configuration is stored and how configuration changes flow through the system.
+
+We'll start from the point where sleds are powered on, even though a lot happens with the rectifiers, service processors, Sidecars, etc. before that point.  Once host systems are powered on:
+
+* Sled agents start up, communicate with each other, and form a trust quorum that enables each of them to decrypt their local storage.  This local storage includes:
+** a **bootstore** containing basic network configuration needed to bring up the rack
+** information about what control plane services are running on this sled
+* Sled agents apply any needed network configuration and start any services they're supposed to be running:
+** On Scrimlets, the switch zone and boundary NTP are started.  Boundary NTP synchronizes time from the customer-provided NTP servers.
+** On non-Scrimlets, internal DNS is started.  The rest of cold boot waits until time has been synchronized from the boundary NTP instances.
+** Once time is synchronized, internal DNS services are started so that components can find each other.
+** Once internal DNS is available, all other services are started concurrently.
+*** CockroachDB nodes start up, discover the rest of the cluster via DNS, and form a cluster.
+*** Nexus starts up and waits for CockroachDB to become available.
+*** All other services start up and wait for their dependencies to become available.
+
+For this to work:
+
+* **Bootstore** must contain enough information to configure networking on the switches and each host to reach other services within the rack as well as the outside world (for NTP).
+* **Internal DNS** must be able to come up without any external dependencies, meaning it stores a complete copy of all DNS data locally.
+
+However, Nexus is the place where all _changes_ to configuration are made, and CockroachDB is the source of truth for all configuration.  As a result, when changing bootstore contents or internal DNS, the change is first made at Nexus, stored into CockroachDB, and then propagated to all sleds and internal DNS instances for local persistent storage so that it's available on cold start (of the _sled_) without the rest of the control plane being up.
+
+This is a very rough approximation, but gives an idea of the dependencies associated with cold start.
+
+[bibliography]
+== References
+
+Unfortunately, most of these RFDs are not yet public.
+
+* [[[rfd4, RFD 4]]] https://4.rfd.oxide.computer/[RFD 4 User Facing API]
+* [[[rfd6, RFD 6]]] https://6.rfd.oxide.computer/[RFD 6 Threat Model]
```

Comment:
The actual title is different.

Reply:
Yeah, RFD 6 is kind of an unfortunate case.  The reference to RFD 6 in this RFD is from a point where the RFD _was_ called "Threat Model" and contained a bunch of useful information about security for the control plane.  At some point the title and contents of RFD 6 were completely changed and that content was lost.  Since it doesn't exist anywhere else, I wanted to leave a breadcrumb here, but there's not a great thing to link to.  I have at least updated this reference to make it less of a dead end.